src/main/java/picard/nio/PicardHtsPath.java

```diff
@@ -117,4 +121,45 @@ public static List<Path> toPaths(final Collection<PicardHtsPath> picardHtsPaths)
         Objects.requireNonNull(picardHtsPaths);
         return picardHtsPaths.stream().map(PicardHtsPath::toPath).collect(Collectors.toList());
     }
+
+    /**
+     * Takes an IOPath and returns a new PicardHtsPath object that keeps the same basename as the original but has
+     * a new extension. If the input path is a relative local path, it is converted to an absolute path via
+     * PicardHtsPath::toPath (we might change this behavior and avoid converting to an absolute path in the future,
+     * by for instance working directly with a string).
```

Comment:
The behavior described here is a bit of an implementation detail, and as you mention here, could be fixed by changing the implementation start with the rawInputString (I'm not suggesting we do that now - since I think as a practical matter this only impacts your tests because you want to compare the old and new paths). Anyway I'd change this comment slightly if you want to retain it:
```suggestion
     * Takes an IOPath and returns a new PicardHtsPath object that keeps the same basename as the original but has
     * a new extension. If the input path was created from a rawInputString that specifies a relative local path, the new path will
     * always have a rawInputString that specifies an absolute path.
```

Reply:
I see that the javadoc is meant for the users of the function and not where the developers describe the implementation details. At the same time I think this is unnatural behavior, and the reason why it's implemented this way should be documented. I think here in the javadoc seems the most fitting place to put this information (the second best place might be the right before we return the output, because that's where this local to absolute path conversion happens under the hood).

Reply:
Sounds good. I incorporated your text.